src/Core/PropertyBag.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.Collections.Concurrent;
+using System.Runtime.CompilerServices;
+
+namespace CodeMonkeys
+{
+    public class PropertyBag
+    {
+        private readonly ConcurrentDictionary<string, object> _properties =
+            new ConcurrentDictionary<string, object>();
+
+        public bool SetValue<TProperty>(
+            TProperty value,
+            [CallerMemberName]string propertyName = "")
+        {
+            var stored = _properties.AddOrUpdate(
+                propertyName,
+                value,
+                (name, oldValue) => value);
```

Comment:
Does it make sense to add a check wether the values differ?

Reply:
Which benefit do you see by doing so?

Reply:
Was thinking about not raising `PropertyChanged` but this is specific for the `BindingBase`, so we can discuss this there.